Observer/AfterConfigSaveObserver.php

```diff
@@ -170,6 +177,42 @@ public function execute(Observer $observer)
         
     }
 
+            /**
+     * Create and track admin form data on Save Config click
+     */
+    public function saveConfigData($razorpayParams)
+    {
+        $storeName = "";
+
+        $firstElement = array_values($razorpayParams)[0];
+        if (empty($firstElement) === false and array_keys($firstElement)[0] === "value")
+        {
+            $razorpayParamsFormattedArray = array('config_settings' => array());
+            foreach($razorpayParams as $key=>$value)
+            {
+                $razorpayParamsFormattedArray['config_settings'][$key] = empty(array_values($value)) === false? 
+                                                                            array_values($value)[0] :
+                                                                            null;
+            }
+            $storeName = $razorpayParamsFormattedArray['config_settings']['merchant_name_override'];
+        }
+
+        $metaData = array("store_name" => $storeName);
+
+        $eventData = array_merge($razorpayParamsFormattedArray, $metaData);
+
+        $this->trackPluginInstrumentation = new TrackPluginInstrumentation(
+                                                $this->config->getConfigData(Config::KEY_PUBLIC_KEY), 
+                                                $this->config->getConfigData(Config::KEY_PRIVATE_KEY),
+                                                $this->moduleList,
+                                                $this->logger
+                                            );
+
+        $response = $this->trackPluginInstrumentation->rzpTrackSegment('Save Config Clicked', ($eventData));
```

Comment:
@nikhilsharma1100 Define variable `$response` only if you are returning values from that function OR sending response to API, please check & remove unused `$response` variables created while calling `rzpTrackSegment` method.

Reply:
I am going to log the response. That's why I am saving it